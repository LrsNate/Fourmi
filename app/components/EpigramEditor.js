import PropTypes from "prop-types";
import React from "react";

import FourmiPropTypes from "../constants/types";

const EpigramEditor = ({ epigram, onSave }) => {
  return (
    <div>
      <p>{epigram.title}</p>
      <button type="button" onClick={onSave}>
        Go back
      </button>
    </div>
  );
};

EpigramEditor.propTypes = {
  epigram: FourmiPropTypes.epigram.isRequired,
  onSave: PropTypes.func.isRequired
};

export default EpigramEditor;
