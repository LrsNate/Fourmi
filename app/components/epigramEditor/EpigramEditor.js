import PropTypes from "prop-types";
import React from "react";
import { Form } from "react-final-form";

import FourmiPropTypes from "../../constants/types";
import TextInput from "../forms/TextInput";

const AttributesEditor = () => {
  return (
    <div>
      <TextInput name="title" />
      <TextInput name="author" />
      <TextInput name="reference" />
    </div>
  );
};

const EpigramEditor = ({ epigram, onSave }) => {
  /**
   * Cards:
   * 1. Info
   * 2. (Latin Text?)
   * 3. French Text
   * 4. Notes
   */
  return (
    <Form initialValues={epigram} onSubmit={onSave}>
      {({ handleSubmit, values }) => (
        <form onSubmit={handleSubmit}>
          <AttributesEditor />
          <button type="button" onClick={() => onSave(values)}>
            Go back
          </button>
        </form>
      )}
    </Form>
  );
};

EpigramEditor.propTypes = {
  epigram: FourmiPropTypes.epigram.isRequired,
  onSave: PropTypes.func.isRequired
};

export default EpigramEditor;
