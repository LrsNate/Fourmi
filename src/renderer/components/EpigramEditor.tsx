import { Card, CardContent, CardHeader, Grid } from "@material-ui/core";
import * as React from "react";
import { Form } from "react-final-form";
import { Epigram } from "../constants/types";
import TextInput from "./forms/TextInput";

interface EpigramEditorProps {
  epigram: Epigram;
  onSave: (values: Epigram) => void;
}

class EpigramEditor extends React.Component<EpigramEditorProps> {
  public handleSave = (values: object) => {
    this.props.onSave(values as Epigram);
  };

  public render() {
    const { epigram } = this.props;

    return (
      <Form initialValues={epigram} onSubmit={this.handleSave}>
        {({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>{this.renderAttributes()}</form>
        )}
      </Form>
    );
  }

  public renderAttributes() {
    return (
      <Card className={"?"}>
        <CardHeader title="Propriétés" />
        <CardContent>
          <Grid container={true}>
            <Grid item={true} sm={3}>
              <TextInput label="Auteur" name="author" />
            </Grid>
            <Grid item={true} sm={3}>
              <TextInput label="Référence" name="reference" />
            </Grid>
            <Grid item={true} sm={6}>
              <TextInput label="Titre" name="title" />
            </Grid>
            <Grid item={true} sm={3}>
              <TextInput label="Mètre" name="meter" />
            </Grid>
            <Grid item={true} sm={3}>
              <TextInput label="Strophe" name="stanza" />
            </Grid>
            <Grid item={true} sm={3}>
              <TextInput label="Destinataire" name="addressee" />
            </Grid>
            <Grid item={true} sm={3}>
              <TextInput label="Date" name="date" />
            </Grid>
            <Grid item={true} sm={12}>
              <TextInput label="Thèmes" name="themes" />
            </Grid>
          </Grid>
        </CardContent>
      </Card>
    );
  }
}

export default EpigramEditor;
