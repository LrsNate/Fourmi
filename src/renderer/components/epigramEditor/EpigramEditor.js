import { Button, Card, CardContent, CardHeader, Grid } from "material-ui";
import PropTypes from "prop-types";
import React from "react";
import { Form } from "react-final-form";

import FourmiPropTypes from "../../constants/types";
import TextInput from "../forms/TextInput";

const AttributesEditor = () => {
  return (
    <Card>
      <CardContent>
        <CardHeader title="Propriétés" />
        <Grid container>
          <Grid item sm={3}>
            <TextInput label="Auteur" name="author" />
          </Grid>
          <Grid item sm={3}>
            <TextInput label="Référence" name="reference" />
          </Grid>
          <Grid item sm={6}>
            <TextInput label="Titre" name="title" />
          </Grid>
          <Grid item sm={3}>
            <TextInput label="Mètre" name="meter" />
          </Grid>
          <Grid item sm={3}>
            <TextInput label="Strophe" name="stanza" />
          </Grid>
          <Grid item sm={3}>
            <TextInput label="Destinataire" name="addressee" />
          </Grid>
          <Grid item sm={3}>
            <TextInput label="Date" name="date" />
          </Grid>
        </Grid>
      </CardContent>
    </Card>
  );
};

const EpigramEditor = ({ className, epigram, onSave }) => {
  /**
   * Cards:
   * 1. Info
   * 2. (Latin Text?)
   * 3. French Text
   * 4. Notes
   */
  return (
    <div className={className}>
      <Form initialValues={epigram} onSubmit={values => onSave(values)}>
        {({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <AttributesEditor />
            <TextInput label="Notes" name="notes" />
            <Button type="submit" raised color="primary">
              Enregistrer
            </Button>
          </form>
        )}
      </Form>
    </div>
  );
};

EpigramEditor.propTypes = {
  className: PropTypes.string,
  epigram: FourmiPropTypes.epigram.isRequired,
  onSave: PropTypes.func.isRequired
};

export default EpigramEditor;
