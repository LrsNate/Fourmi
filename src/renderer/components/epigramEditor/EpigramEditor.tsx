import {
  Button,
  Card,
  CardContent,
  CardHeader,
  Grid,
  StyleRulesCallback,
  Theme,
  withStyles
} from "@material-ui/core";
import { Save } from "@material-ui/icons";
import * as React from "react";
import { Field, Form } from "react-final-form";
import { RouteComponentProps, withRouter } from "react-router";
import { Epigram } from "../../constants/types";
import { searchRoutePath } from "../../routes";
import TextEditor from "../forms/TextEditor";
import TextInput from "../forms/TextInput";
import ChipEditorCard from "./ChipEditorCard";

const styles: StyleRulesCallback<string> = (theme: Theme) => ({
  saveButton: {
    position: "fixed",
    right: theme.spacing.unit * 2,
    bottom: theme.spacing.unit * 2,
    zIndex: 1
  }
});

interface EpigramEditorProps extends RouteComponentProps<{}> {
  epigram: Epigram;
  onSave: (values: Epigram) => void;
  classes: Record<string, string>;
}

class EpigramEditor extends React.Component<EpigramEditorProps> {
  public handleSave = (values: object) => {
    const { history, onSave } = this.props;
    onSave(values as Epigram);
    history.push(searchRoutePath());
  };

  public render() {
    const { classes, epigram } = this.props;

    return (
      <Form initialValues={epigram} onSubmit={this.handleSave}>
        {({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <Grid container direction="column" spacing={8}>
              <Grid item>{this.renderAttributes()}</Grid>
              <Grid item>{this.renderThemes()}</Grid>
              <Grid item>
                {this.renderTextEditor("Texte latin", "latinText")}
              </Grid>
              <Grid item>
                {this.renderTextEditor("Texte français", "frenchText")}
              </Grid>
              <Grid item>{this.renderTextEditor("Notes", "notes")}</Grid>
            </Grid>
            <Button
              className={classes.saveButton}
              type="submit"
              variant="fab"
              color="secondary"
            >
              <Save />
            </Button>
          </form>
        )}
      </Form>
    );
  }

  public renderAttributes() {
    return (
      <Card>
        <CardHeader title="Propriétés" />
        <CardContent>
          <Grid container spacing={8}>
            <Grid item sm={3}>
              <TextInput label="Auteur" name="author" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Référence" name="reference" />
            </Grid>
            <Grid item sm={6}>
              <TextInput label="Titre" name="title" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Mètre" name="meter" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Strophe" name="stanza" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Destinataire" name="addressee" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Date" name="date" />
            </Grid>
          </Grid>
        </CardContent>
      </Card>
    );
  }

  public renderThemes() {
    return (
      <Field name="themes">
        {({ input: { onChange, value } }) => (
          <ChipEditorCard values={value} onChange={onChange} />
        )}
      </Field>
    );
  }

  public renderTextEditor(label: string, name: string) {
    return (
      <Card>
        <CardHeader title={label} />
        <CardContent>
          <TextEditor name={name} />
        </CardContent>
      </Card>
    );
  }
}

export default withStyles(styles)(withRouter(EpigramEditor));
