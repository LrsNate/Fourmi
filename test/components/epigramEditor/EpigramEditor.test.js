import { render } from "enzyme";
import React from "react";

import EpigramEditor from "../../../app/components/epigramEditor/EpigramEditor";

describe("The EpigramEditor component", () => {
  describe("on an initial load", () => {
    it("matches the snapshot", () => {
      const epigram = {};
      const wrapper = render(
        <EpigramEditor epigram={epigram} onSave={() => {}} />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
