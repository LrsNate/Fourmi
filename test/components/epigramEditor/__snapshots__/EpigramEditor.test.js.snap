// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The EpigramEditor component on an initial load matches the snapshot 1`] = `
<form>
  <div
    class="MuiPaper-root-2 MuiPaper-shadow2-6 MuiPaper-rounded-3 EpigramEditor-card-1"
  >
    <div
      class="MuiCardContent-root-35 MuiCardHeader-root-29"
    >
      <div
        class="MuiCardHeader-content-32"
      >
        <span
          class="MuiTypography-root-36 MuiTypography-headline-41 MuiCardHeader-title-33"
        >
          Propriétés
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root-35"
    >
      <div
        class="MuiGrid-typeContainer-60 MuiGrid-spacing-xs-16-81"
      >
        <div
          class="MuiGrid-typeItem-61 MuiGrid-grid-sm-3-100"
        >
          <div
            class="MuiFormControl-root-149 MuiFormControl-marginNormal-150 MuiFormControl-fullWidth-152"
          >
            <label
              class="MuiFormLabel-root-159 MuiInputLabel-root-153 MuiInputLabel-formControl-154 MuiInputLabel-animated-157"
              data-shrink="false"
            >
              Auteur
            </label>
            <div
              class="MuiInput-root-163 MuiInput-formControl-164 MuiInput-inkbar-165 MuiInput-underline-169"
            >
              <input
                class="MuiInput-input-172 MuiInput-inputSingleline-175"
                name="author"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-61 MuiGrid-grid-sm-3-100"
        >
          <div
            class="MuiFormControl-root-149 MuiFormControl-marginNormal-150 MuiFormControl-fullWidth-152"
          >
            <label
              class="MuiFormLabel-root-159 MuiInputLabel-root-153 MuiInputLabel-formControl-154 MuiInputLabel-animated-157"
              data-shrink="false"
            >
              Référence
            </label>
            <div
              class="MuiInput-root-163 MuiInput-formControl-164 MuiInput-inkbar-165 MuiInput-underline-169"
            >
              <input
                class="MuiInput-input-172 MuiInput-inputSingleline-175"
                name="reference"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-61 MuiGrid-grid-sm-6-103"
        >
          <div
            class="MuiFormControl-root-149 MuiFormControl-marginNormal-150 MuiFormControl-fullWidth-152"
          >
            <label
              class="MuiFormLabel-root-159 MuiInputLabel-root-153 MuiInputLabel-formControl-154 MuiInputLabel-animated-157"
              data-shrink="false"
            >
              Titre
            </label>
            <div
              class="MuiInput-root-163 MuiInput-formControl-164 MuiInput-inkbar-165 MuiInput-underline-169"
            >
              <input
                class="MuiInput-input-172 MuiInput-inputSingleline-175"
                name="title"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-61 MuiGrid-grid-sm-3-100"
        >
          <div
            class="MuiFormControl-root-149 MuiFormControl-marginNormal-150 MuiFormControl-fullWidth-152"
          >
            <label
              class="MuiFormLabel-root-159 MuiInputLabel-root-153 MuiInputLabel-formControl-154 MuiInputLabel-animated-157"
              data-shrink="false"
            >
              Mètre
            </label>
            <div
              class="MuiInput-root-163 MuiInput-formControl-164 MuiInput-inkbar-165 MuiInput-underline-169"
            >
              <input
                class="MuiInput-input-172 MuiInput-inputSingleline-175"
                name="meter"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-61 MuiGrid-grid-sm-3-100"
        >
          <div
            class="MuiFormControl-root-149 MuiFormControl-marginNormal-150 MuiFormControl-fullWidth-152"
          >
            <label
              class="MuiFormLabel-root-159 MuiInputLabel-root-153 MuiInputLabel-formControl-154 MuiInputLabel-animated-157"
              data-shrink="false"
            >
              Strophe
            </label>
            <div
              class="MuiInput-root-163 MuiInput-formControl-164 MuiInput-inkbar-165 MuiInput-underline-169"
            >
              <input
                class="MuiInput-input-172 MuiInput-inputSingleline-175"
                name="stanza"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-61 MuiGrid-grid-sm-3-100"
        >
          <div
            class="MuiFormControl-root-149 MuiFormControl-marginNormal-150 MuiFormControl-fullWidth-152"
          >
            <label
              class="MuiFormLabel-root-159 MuiInputLabel-root-153 MuiInputLabel-formControl-154 MuiInputLabel-animated-157"
              data-shrink="false"
            >
              Destinataire
            </label>
            <div
              class="MuiInput-root-163 MuiInput-formControl-164 MuiInput-inkbar-165 MuiInput-underline-169"
            >
              <input
                class="MuiInput-input-172 MuiInput-inputSingleline-175"
                name="addressee"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-61 MuiGrid-grid-sm-3-100"
        >
          <div
            class="MuiFormControl-root-149 MuiFormControl-marginNormal-150 MuiFormControl-fullWidth-152"
          >
            <label
              class="MuiFormLabel-root-159 MuiInputLabel-root-153 MuiInputLabel-formControl-154 MuiInputLabel-animated-157"
              data-shrink="false"
            >
              Date
            </label>
            <div
              class="MuiInput-root-163 MuiInput-formControl-164 MuiInput-inkbar-165 MuiInput-underline-169"
            >
              <input
                class="MuiInput-input-172 MuiInput-inputSingleline-175"
                name="date"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-61 MuiGrid-grid-sm-12-109"
        >
          <div
            class="MuiFormControl-root-149 MuiFormControl-marginNormal-150 MuiFormControl-fullWidth-152"
          >
            <label
              class="MuiFormLabel-root-159 MuiInputLabel-root-153 MuiInputLabel-formControl-154 MuiInputLabel-animated-157"
              data-shrink="false"
            >
              Vices
            </label>
            <div
              class="MuiInput-root-163 MuiInput-formControl-164 MuiInput-inkbar-165 MuiInput-underline-169"
            >
              <input
                class="MuiInput-input-172 MuiInput-inputSingleline-175"
                name="vices"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-2 MuiPaper-shadow2-6 MuiPaper-rounded-3 EpigramEditor-card-1"
  >
    <div
      class="MuiCardContent-root-35 MuiCardHeader-root-29"
    >
      <div
        class="MuiCardHeader-content-32"
      >
        <span
          class="MuiTypography-root-36 MuiTypography-headline-41 MuiCardHeader-title-33"
        >
          Texte latin
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root-35"
    >
      <div>
        Rich text editor mock
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-2 MuiPaper-shadow2-6 MuiPaper-rounded-3 EpigramEditor-card-1"
  >
    <div
      class="MuiCardContent-root-35 MuiCardHeader-root-29"
    >
      <div
        class="MuiCardHeader-content-32"
      >
        <span
          class="MuiTypography-root-36 MuiTypography-headline-41 MuiCardHeader-title-33"
        >
          Texte français
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root-35"
    >
      <div>
        Rich text editor mock
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-2 MuiPaper-shadow2-6 MuiPaper-rounded-3 EpigramEditor-card-1"
  >
    <div
      class="MuiCardContent-root-35 MuiCardHeader-root-29"
    >
      <div
        class="MuiCardHeader-content-32"
      >
        <span
          class="MuiTypography-root-36 MuiTypography-headline-41 MuiCardHeader-title-33"
        >
          Notes
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root-35"
    >
      <div>
        Rich text editor mock
      </div>
    </div>
  </div>
  <button
    class="MuiButtonBase-root-195 MuiButton-root-180 MuiButton-raised-187 MuiButton-raisedPrimary-189"
    role="button"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiButton-label-182"
    >
      Enregistrer
    </span>
    <span
      class="MuiTouchRipple-root-197"
    />
  </button>
</form>
`;
